moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -1062,6 +1075,55 @@ bool PlanningSceneMonitor::getShapeTransformCache(const std::string& target_fram
   return true;
 }
 
+UniqueTimedLock PlanningSceneMonitor::acquireUniqueLock(const std::string& source, std::chrono::milliseconds timeout)
+{
+  // TODO(henningkayser): lock/unlock octomap
```

Comment:
The octomap could be locked here and then unlocked with the destructor. However, this would increase the time the octree is being locked since some callbacks haven't done this before.